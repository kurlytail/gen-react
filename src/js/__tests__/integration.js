import { execSync } from 'child_process';

describe('# integration test', () => {
    beforeEach(() => {
        execSync('rm -rf testoutput');
    });

    it('## should print help options', () => {
        const output = execSync('./scripts/sgen-react.sh -h').toString();
        expect(output).toMatchSnapshot();
    });

    it('## should generate design', () => {
        const output = execSync('./scripts/sgen-react.sh -d src/test/fixture/design.json -o testoutput').toString();
        expect(output).toMatchSnapshot();
    });

    it('## should generate design with merge', () => {
        let output = execSync('./scripts/sgen-react.sh -d src/test/fixture/design.json -o testoutput').toString();
        expect(output).toMatchSnapshot();
        output = execSync('./scripts/sgen-react.sh -d src/test/fixture/design.json -o testoutput').toString();
        expect(output).toMatchSnapshot();
    });

    it('## should generate design and run npm commands', () => {
        let output = execSync('./scripts/sgen-react.sh -d src/test/fixture/design.json -o testoutput').toString();
        expect(output).toMatchSnapshot();
        output = execSync('npm install', { cwd: 'testoutput' }).toString();
        output = execSync('npm run lint', { cwd: 'testoutput' }).toString();
        output = execSync('npm run build', { cwd: 'testoutput' }).toString();
        output = execSync('npm test -- -u', { cwd: 'testoutput' }).toString();
    });

    it('## should generate design with bootstrap extensions and run npm commands', () => {
        let output = execSync(
            './scripts/sgen-react.sh -e templates/addons/bootstrap4 -d src/test/fixture/design.json -o testoutput'
        ).toString();
        expect(output).toMatchSnapshot();
        output = execSync('npm install', { cwd: 'testoutput' }).toString();
        output = execSync('npm run lint', { cwd: 'testoutput' }).toString();
        output = execSync('npm run build', { cwd: 'testoutput' }).toString();
        output = execSync('npm test -- -u', { cwd: 'testoutput' }).toString();
    });
});
