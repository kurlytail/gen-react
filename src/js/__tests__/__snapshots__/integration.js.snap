// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# integration test ## should generate design 1`] = `
"info: loading map node_modules/@kurlytail/gen-npm/templates/map.json
info: loading map templates/map.json
info: Initializing git repository testoutput/.git
info: Generated testoutput/.editorconfig from template node_modules/@kurlytail/gen-npm/templates/.editorconfig._
info: Generated testoutput/.eslintignore from template node_modules/@kurlytail/gen-npm/templates/.eslintignore._
info: loading extension templates/extensions/babelrc._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/babelrc._
info: Generated testoutput/.babelrc from template node_modules/@kurlytail/gen-npm/templates/.babelrc._
info: Generated testoutput/.eslintrc.js from template node_modules/@kurlytail/gen-npm/templates/.eslintrc.js._
info: Generated testoutput/.gitignore from template node_modules/@kurlytail/gen-npm/templates/.gitignore._
info: Generated testoutput/.npmignore from template node_modules/@kurlytail/gen-npm/templates/.npmignore._
info: Generated testoutput/.prettierrc from template node_modules/@kurlytail/gen-npm/templates/.prettierrc._
info: Generated testoutput/Jenkinsfile from template node_modules/@kurlytail/gen-npm/templates/Jenkinsfile._
info: loading extension templates/extensions/package-json-react._
info: loading extension templates/extensions/package-json-jsdoc._
info: loading extension templates/extensions/package-json-client._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/package-json-misc._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/package-json-jest._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/package-json-babel._
info: Generated testoutput/package.json from template node_modules/@kurlytail/gen-npm/templates/package.json._
info: Generated testoutput/README.md from template node_modules/@kurlytail/gen-npm/templates/README.md._
info: loading extension templates/extensions/webpack-config._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/webpack-config._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/webpack-config-1._
info: Generated testoutput/webpack.config.js from template node_modules/@kurlytail/gen-npm/templates/webpack.config.js._
info: Generated testoutput/src/js/index.js from template templates/src/js/index.js._
info: Generated testoutput/src/js/__tests__/integration.js from template node_modules/@kurlytail/gen-npm/templates/integration.js._
info: Generated testoutput/jsdoc.json from template templates/jsdoc.json._
info: Generated testoutput/src/js/components/PasswordComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/PasswordComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/RegisterComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/RegisterComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/InputComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/InputComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/LoginComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/LoginComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/LinkComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/LinkComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/CheckboxComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/CheckboxComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/ButtonComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/ButtonComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/index.js from template templates/src/js/components/index.js._
info: Generated testoutput/src/js/html.jst from template templates/src/js/html.jst._
info: Committed generated files
info: Attempting to use branch master-sgen-generated
info: Creating new branch master-sgen-generated
warn: Checkout branch master
warn: Checkout branch master
info: sgen is done
"
`;

exports[`# integration test ## should generate design and run react commands 1`] = `
"info: loading map node_modules/@kurlytail/gen-npm/templates/map.json
info: loading map templates/map.json
info: Initializing git repository testoutput/.git
info: Generated testoutput/.editorconfig from template node_modules/@kurlytail/gen-npm/templates/.editorconfig._
info: Generated testoutput/.eslintignore from template node_modules/@kurlytail/gen-npm/templates/.eslintignore._
info: loading extension templates/extensions/babelrc._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/babelrc._
info: Generated testoutput/.babelrc from template node_modules/@kurlytail/gen-npm/templates/.babelrc._
info: Generated testoutput/.eslintrc.js from template node_modules/@kurlytail/gen-npm/templates/.eslintrc.js._
info: Generated testoutput/.gitignore from template node_modules/@kurlytail/gen-npm/templates/.gitignore._
info: Generated testoutput/.npmignore from template node_modules/@kurlytail/gen-npm/templates/.npmignore._
info: Generated testoutput/.prettierrc from template node_modules/@kurlytail/gen-npm/templates/.prettierrc._
info: Generated testoutput/Jenkinsfile from template node_modules/@kurlytail/gen-npm/templates/Jenkinsfile._
info: loading extension templates/extensions/package-json-react._
info: loading extension templates/extensions/package-json-jsdoc._
info: loading extension templates/extensions/package-json-client._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/package-json-misc._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/package-json-jest._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/package-json-babel._
info: Generated testoutput/package.json from template node_modules/@kurlytail/gen-npm/templates/package.json._
info: Generated testoutput/README.md from template node_modules/@kurlytail/gen-npm/templates/README.md._
info: loading extension templates/extensions/webpack-config._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/webpack-config._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/webpack-config-1._
info: Generated testoutput/webpack.config.js from template node_modules/@kurlytail/gen-npm/templates/webpack.config.js._
info: Generated testoutput/src/js/index.js from template templates/src/js/index.js._
info: Generated testoutput/src/js/__tests__/integration.js from template node_modules/@kurlytail/gen-npm/templates/integration.js._
info: Generated testoutput/jsdoc.json from template templates/jsdoc.json._
info: Generated testoutput/src/js/components/PasswordComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/PasswordComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/RegisterComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/RegisterComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/InputComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/InputComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/LoginComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/LoginComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/LinkComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/LinkComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/CheckboxComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/CheckboxComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/ButtonComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/ButtonComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/index.js from template templates/src/js/components/index.js._
info: Generated testoutput/src/js/html.jst from template templates/src/js/html.jst._
info: Committed generated files
info: Attempting to use branch master-sgen-generated
info: Creating new branch master-sgen-generated
warn: Checkout branch master
warn: Checkout branch master
info: sgen is done
"
`;

exports[`# integration test ## should generate design with merge 1`] = `
"info: loading map node_modules/@kurlytail/gen-npm/templates/map.json
info: loading map templates/map.json
info: Initializing git repository testoutput/.git
info: Generated testoutput/.editorconfig from template node_modules/@kurlytail/gen-npm/templates/.editorconfig._
info: Generated testoutput/.eslintignore from template node_modules/@kurlytail/gen-npm/templates/.eslintignore._
info: loading extension templates/extensions/babelrc._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/babelrc._
info: Generated testoutput/.babelrc from template node_modules/@kurlytail/gen-npm/templates/.babelrc._
info: Generated testoutput/.eslintrc.js from template node_modules/@kurlytail/gen-npm/templates/.eslintrc.js._
info: Generated testoutput/.gitignore from template node_modules/@kurlytail/gen-npm/templates/.gitignore._
info: Generated testoutput/.npmignore from template node_modules/@kurlytail/gen-npm/templates/.npmignore._
info: Generated testoutput/.prettierrc from template node_modules/@kurlytail/gen-npm/templates/.prettierrc._
info: Generated testoutput/Jenkinsfile from template node_modules/@kurlytail/gen-npm/templates/Jenkinsfile._
info: loading extension templates/extensions/package-json-react._
info: loading extension templates/extensions/package-json-jsdoc._
info: loading extension templates/extensions/package-json-client._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/package-json-misc._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/package-json-jest._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/package-json-babel._
info: Generated testoutput/package.json from template node_modules/@kurlytail/gen-npm/templates/package.json._
info: Generated testoutput/README.md from template node_modules/@kurlytail/gen-npm/templates/README.md._
info: loading extension templates/extensions/webpack-config._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/webpack-config._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/webpack-config-1._
info: Generated testoutput/webpack.config.js from template node_modules/@kurlytail/gen-npm/templates/webpack.config.js._
info: Generated testoutput/src/js/index.js from template templates/src/js/index.js._
info: Generated testoutput/src/js/__tests__/integration.js from template node_modules/@kurlytail/gen-npm/templates/integration.js._
info: Generated testoutput/jsdoc.json from template templates/jsdoc.json._
info: Generated testoutput/src/js/components/PasswordComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/PasswordComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/RegisterComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/RegisterComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/InputComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/InputComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/LoginComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/LoginComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/LinkComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/LinkComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/CheckboxComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/CheckboxComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/ButtonComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/ButtonComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/index.js from template templates/src/js/components/index.js._
info: Generated testoutput/src/js/html.jst from template templates/src/js/html.jst._
info: Committed generated files
info: Attempting to use branch master-sgen-generated
info: Creating new branch master-sgen-generated
warn: Checkout branch master
warn: Checkout branch master
info: sgen is done
"
`;

exports[`# integration test ## should generate design with merge 2`] = `
"info: loading map node_modules/@kurlytail/gen-npm/templates/map.json
info: loading map templates/map.json
info: Opening git repository testoutput/.git
warn: Ignoring error Error: cannot stash changes - there is nothing to stash.
info: Attempting to use branch master-sgen-generated
warn: Checkout branch master-sgen-generated
info: Generated testoutput/.editorconfig from template node_modules/@kurlytail/gen-npm/templates/.editorconfig._
info: Generated testoutput/.eslintignore from template node_modules/@kurlytail/gen-npm/templates/.eslintignore._
info: loading extension templates/extensions/babelrc._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/babelrc._
info: Generated testoutput/.babelrc from template node_modules/@kurlytail/gen-npm/templates/.babelrc._
info: Generated testoutput/.eslintrc.js from template node_modules/@kurlytail/gen-npm/templates/.eslintrc.js._
info: Generated testoutput/.gitignore from template node_modules/@kurlytail/gen-npm/templates/.gitignore._
info: Generated testoutput/.npmignore from template node_modules/@kurlytail/gen-npm/templates/.npmignore._
info: Generated testoutput/.prettierrc from template node_modules/@kurlytail/gen-npm/templates/.prettierrc._
info: Generated testoutput/Jenkinsfile from template node_modules/@kurlytail/gen-npm/templates/Jenkinsfile._
info: loading extension templates/extensions/package-json-react._
info: loading extension templates/extensions/package-json-jsdoc._
info: loading extension templates/extensions/package-json-client._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/package-json-misc._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/package-json-jest._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/package-json-babel._
info: Generated testoutput/package.json from template node_modules/@kurlytail/gen-npm/templates/package.json._
info: Generated testoutput/README.md from template node_modules/@kurlytail/gen-npm/templates/README.md._
info: loading extension templates/extensions/webpack-config._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/webpack-config._
info: loading extension node_modules/@kurlytail/gen-npm/templates/extensions/webpack-config-1._
info: Generated testoutput/webpack.config.js from template node_modules/@kurlytail/gen-npm/templates/webpack.config.js._
info: Generated testoutput/src/js/index.js from template templates/src/js/index.js._
info: Generated testoutput/src/js/__tests__/integration.js from template node_modules/@kurlytail/gen-npm/templates/integration.js._
info: Generated testoutput/jsdoc.json from template templates/jsdoc.json._
info: Generated testoutput/src/js/components/PasswordComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/PasswordComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/RegisterComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/RegisterComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/InputComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/InputComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/LoginComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/LoginComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/LinkComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/LinkComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/CheckboxComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/CheckboxComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/ButtonComponent.jsx from template templates/src/js/components/component.jsx._
info: Generated testoutput/src/js/components/__tests__/ButtonComponent.jsx from template templates/src/js/components/__tests__/component.jsx._
info: Generated testoutput/src/js/components/index.js from template templates/src/js/components/index.js._
info: Generated testoutput/src/js/html.jst from template templates/src/js/html.jst._
warn: Checkout branch master-sgen-generated
warn: Checkout branch master-sgen-generated
warn: Nothing changed in generated files
warn: Checkout branch master
info: sgen is done
"
`;

exports[`# integration test ## should print help options 1`] = `
"Usage:
  node sgen [OPTION]

Options:
  -m, --map=ARG+       mapping files
  -d, --design=ARG+    design files
  -e, --extension=ARG+ extension directories or files
  -o, --output=ARG     output directory for generated files
      --overwrite=ARG  Overwrite modes, can be one of skip, regen, error                                  (default: error)
      --forceOverwrite Override overwrite flags in the generator map with the one provided on --overwrite
  -h, --help           display this help
"
`;
